src/GraphQL/Utilities/LinkConfiguration.cs

```diff
@@ -0,0 +1,318 @@
+using System.Collections;
+using GraphQL.Types;
+using GraphQL.Types.Scalars;
+
+namespace GraphQL.Utilities;
+
+/// <summary>
+/// Configuration for the @link directive, allowing customization of the linked schema.
+/// </summary>
+public sealed class LinkConfiguration
+{
+    internal const string LINK_URL = "https://specs.apollo.dev/link/v1.0";
+    internal const string METADATA_KEY = "__Link_Configuration";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LinkConfiguration"/> class with the specified URL.
+    /// </summary>
+    public LinkConfiguration(string url)
+    {
+        Url = url ?? throw new ArgumentNullException(nameof(url));
+        var (name, _) = TryParseUrl(url);
+        _namespace = _originalNamespace = name;
+    }
+
+    /// <summary>
+    /// The URL of the linked schema.
+    /// </summary>
+    public string Url { get; }
+
+    private readonly string? _originalNamespace;
+
+    private string? _namespace;
+    /// <summary>
+    /// The namespace that imported elements are placed in when they are not specifically imported.
+    /// Defaults to the name of the imported schema, as derived from the URL.
+    /// </summary>
+    /// <remarks>
+    /// For example, the namespace "foo" would import the type "Bar" from the linked schema as "foo__Bar"
+    /// when "Bar" is not specifically imported.
+    /// However, importing the type "Bar" would allow it to be used as "Bar" without a namespace prefix.
+    /// </remarks>
+    public string? Namespace
+    {
+        get => _namespace;
+        set
+        {
+            if (value == null && _originalNamespace != null)
+                throw new InvalidOperationException("Cannot set the namespace to null when it can be derived from the URL.");
+            if (value == "")
+                throw new InvalidOperationException("The default namespace prefix cannot be an empty string.");
+            if (value != null)
+                NameValidator.ValidateName(value, NamedElement.Type);
+            _namespace = value;
+        }
+    }
+
+    /// <summary>
+    /// The purpose of the link, which can influence how the link is treated in the schema.
+    /// </summary>
+    public LinkPurpose? Purpose { get; set; }
+
+    /// <summary>
+    /// A dictionary of types and directives to import from the linked schema, specifying the alias to use for each imported member.
+    /// To use the imported member without an alias, specify the same name for the key and value.
+    /// </summary>
+    /// <remarks>
+    /// For example:
+    /// <code>
+    /// link.Imports.Add("@key", "@key");                 // imports @key without an alias
+    /// link.Imports.Add("@shareable", "@fed_shareable"); // imports @shareable as @fed_shareable
```

Comment:
What if I import it with a double underscore `"@fed__shareable"`? Is it legal?

Reply:
Yes, as far as I can tell, it is legal.